src/getstream/updateBetterChannelMembers.js

```diff
@@ -0,0 +1,176 @@
+const _ = require('lodash');
+const {Sequelize} = require('sequelize');
+const {sequelize} = require('../databases/models');
+const BetterSocialConstantListUtils = require('../utils/constantList/utils');
+
+const SEPARATOR = ', ';
+
+const getAllChatAnonimityUserInfo = async (channelId, userIds = []) => {
+  if (!channelId) throw new Error('Channel ID is required');
+
+  if (userIds?.length === 0) return [];
+  const query = `
+    SELECT
+        users.user_id,
+        users.username,
+        users.is_anonymous,
+        chat_anon_user_info.*
+    FROM users
+    LEFT JOIN chat_anon_user_info
+        ON users.user_id = CAST(chat_anon_user_info.my_anon_user_id as varchar)
+        AND chat_anon_user_info.channel_id = :channelId
+    WHERE users.user_id IN (:userIds)`;
+
+  try {
+    const result = await sequelize.query(query, {
+      replacements: {
+        channelId,
+        userIds
+      },
+      type: Sequelize.QueryTypes.SELECT
+    });
+
+    return result;
+  } catch (e) {
+    console.error('Failed to get chat anonimity user info', e);
+    throw e;
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getAllChatAnonimityUserInfo` is using raw SQL query to fetch data from the database. While this is not inherently bad, it can lead to potential SQL injection attacks if not handled properly. In this case, you are using parameterized queries which is a good practice to prevent SQL injection. However, Sequelize provides a way to write these queries in a more secure and readable manner using its own API. This will also make your code more maintainable and consistent.

```diff
-  const query = `
-    SELECT
-        users.user_id,
-        users.username,
-        users.is_anonymous,
-        chat_anon_user_info.*
-    FROM users
-    LEFT JOIN chat_anon_user_info
-        ON users.user_id = CAST(chat_anon_user_info.my_anon_user_id as varchar)
-        AND chat_anon_user_info.channel_id = :channelId
-    WHERE users.user_id IN (:userIds)`;
-
-  try {
-    const result = await sequelize.query(query, {
-      replacements: {
-        channelId,
-        userIds
-      },
-      type: Sequelize.QueryTypes.SELECT
-    });
-
-    return result;
-  } catch (e) {
-    console.error('Failed to get chat anonimity user info', e);
-    throw e;
-  }
+  try {
+    const result = await User.findAll({
+      where: {
+        user_id: {
+          [Sequelize.Op.in]: userIds
+        }
+      },
+      include: [{
+        model: ChatAnonUserInfo,
+        where: { channel_id: channelId },
+        required: false
+      }]
+    });
+
+    return result.map(user => user.get({ plain: true }));
+  } catch (e) {
+    console.error('Failed to get chat anonimity user info', e);
+    throw e;
+  }
```
Please note that you need to import the `User` and `ChatAnonUserInfo` models at the top of your file.



<!-- This is an auto-generated comment by OSS CodeRabbit -->